In [ ]:
import pandas as pd
import numpy as np

In [ ]:
url = 'https://raw.githubusercontent.com/andypilapil/csci-113-lab-1/main/food_coded.csv'
df = pd.read_csv(url)

In [ ]:
df.tail(10)

GPA  Gender  breakfast  calories_chicken  calories_day  calories_scone  \
115    3.3       2          1               610           4.0           980.0   
116    3.4       1          1               610           NaN           420.0   
117   3.77       1          1               610           NaN           315.0   
118   3.63       1          1               430           3.0           420.0   
119    3.2       2          1               610           3.0           420.0   
120    3.5       1          1               610           4.0           420.0   
121      3       1          1               265           2.0           315.0   
122  3.882       1          1               720           NaN           420.0   
123      3       2          1               720           4.0           420.0   
124    3.9       1          1               430           NaN           315.0   

     coffee                                       comfort_food  \
115       2  chocolate bar, ice cream, pretzels, potato chi...   
116       2             Ice cream, chocolate, pizza, cucumber    
117       2  Noodle ( any kinds of noodle), Tuna sandwich, ...   
118       1                               Chinese, chips, cake   
119       2                        chips, rice, chicken curry,   
120       2            wine. mac and cheese, pizza, ice cream    
121       2                         Pizza / Wings / Cheesecake   
122       1                         rice, potato, seaweed soup   
123       1                       Mac n Cheese, Lasagna, Pizza   
124       2                       Chocolates, pizza, and Ritz.   

                                  comfort_food_reasons  \
115              Stress, boredom and physical activity   
116                     loneliness, homework, boredom    
117  When i'm  eating with my close friends/ Food s...   
118                                Stress and boredom    
119                   Happiness, boredom, social event   
120                               boredom and sadness    
121                    Loneliness / Homesick / Sadness   
122                                            sadness   
123      happiness, they are some of my favorite foods   
124                   hormones, Premenstrual syndrome.   

     comfort_food_reasons_coded  ...  soup  sports  thai_food  \
115                         NaN  ...   1.0     1.0          1   
116                         NaN  ...   1.0     2.0          5   
117                         NaN  ...   1.0     2.0          5   
118                         NaN  ...   1.0     2.0          4   
119                         NaN  ...   1.0     1.0          5   
120                         NaN  ...   1.0     1.0          5   
121                         NaN  ...   1.0     NaN          4   
122                         NaN  ...   1.0     2.0          5   
123                         NaN  ...   2.0     2.0          1   
124                         NaN  ...   1.0     2.0          2   

    tortilla_calories  turkey_calories              type_sports veggies_day  \
115            1165.0              690                   Hockey           2   
116             725.0              345                     none           5   
117             725.0              690  No, I don't play sport.           3   
118             940.0              345                     None           5   
119            1165.0              690                   Soccer           5   
120             940.0              500                 Softball           5   
121             940.0              500              basketball            5   
122             580.0              690                     none           4   
123             940.0              500                      NaN           3   
124             725.0              345                      NaN           4   

     vitamins  waffle_calories  weight  
115         2             1315     150  
116         1             1315     170  
117         1              760     113  

In [ ]:
df.columns

Index(['GPA', 'Gender', 'breakfast', 'calories_chicken', 'calories_day',
       'calories_scone', 'coffee', 'comfort_food', 'comfort_food_reasons',
       'comfort_food_reasons_coded', 'cook', 'comfort_food_reasons_coded.1',
       'cuisine', 'diet_current', 'diet_current_coded', 'drink',
       'eating_changes', 'eating_changes_coded', 'eating_changes_coded1',
       'eating_out', 'employment', 'ethnic_food', 'exercise',
       'father_education', 'father_profession', 'fav_cuisine',
       'fav_cuisine_coded', 'fav_food', 'food_childhood', 'fries', 'fruit_day',
       'grade_level', 'greek_food', 'healthy_feeling', 'healthy_meal',
       'ideal_diet', 'ideal_diet_coded', 'income', 'indian_food',
       'italian_food', 'life_rewarding', 'marital_status',
       'meals_dinner_friend', 'mother_education', 'mother_profession',
       'nutritional_check', 'on_off_campus', 'parents_cook', 'pay_meal_out',
       'persian_food', 'self_perception_weight', 'soup', 'sports', 'thai_food',
       

First, we look for columns with null values and clean them. Then, we create a list of columns which have missing values followed by the number of missing values per column.

In [ ]:
cols_with_missing = [col for col in df.columns
                     if df[col].isnull().any()]

In [ ]:
for i in cols_with_missing:
    print(i,df[i].isnull().sum())

GPA 2
calories_day 19
calories_scone 1
comfort_food 1
comfort_food_reasons 1
comfort_food_reasons_coded 19
cook 3
cuisine 17
diet_current 1
drink 2
eating_changes 3
employment 9
exercise 13
father_education 1
father_profession 3
fav_cuisine 2
fav_food 2
food_childhood 1
healthy_meal 1
ideal_diet 1
income 1
life_rewarding 1
marital_status 1
meals_dinner_friend 3
mother_education 3
mother_profession 2
on_off_campus 1
persian_food 1
self_perception_weight 1
soup 1
sports 2
tortilla_calories 1
type_sports 21
weight 2


In [ ]:
df.shape

(125, 61)

**We currently have 125 rows and 61**

Let us now take a closer look at the data and try to replace the missing values using fillna() function. We filter down the columns with missing values that can be replaced with values that is appropriate for them.

In [ ]:
df['calories_day'].fillna(1,inplace=True) # 1 - i dont know how many calories i should consume
df['comfort_food_reasons_coded'].fillna(9,inplace=True) # 9 – none
df['cuisine'].fillna(6,inplace=True) # 6 – other
df['employment'].fillna(4,inplace=True) # 4  - other
df['exercise'].fillna(5,inplace=True) # 5 – Never
df['type_sports'].fillna('Nothing',inplace=True)

After replacing some of the missing values with appropriate values, we remove the rows with the rest of the missing values that can't be replaced.

In [ ]:
for i in cols_with_missing:
    df = df[~df[i].isnull()]

In [ ]:
df.shape

(101, 61)

**We now have 106 rows and 61 columns left**

**GPA Column**

The GPA column contains float values in string format along with missing values and erroneous values. To address this, let us create a new column by first removing the '.'(decimal point) from the values, in order to make use of the .isdigit() function. This .isdigit() function helps you to identify if all the values in the string are digits.

In [ ]:
df['GPA'].value_counts()

3.5           10
3             10
3.3            9
3.2            9
3.7            8
3.4            7
3.6            7
3.8            6
2.8            4
3.9            4
4              3
3.1            2
3.68           1
3.92           1
2.6            1
3.83           1
3.75           1
2.4            1
Unknown        1
3.73           1
3.77           1
3.63           1
3.79 bitch     1
3.89           1
3.67           1
Personal       1
3.35           1
3.605          1
2.9            1
3.654          1
3.65           1
3.904          1
2.25           1
3.882          1
Name: GPA, dtype: int64

In [ ]:
df.dropna(subset=['GPA'],inplace=True)
df['GPA_new'] = df['GPA'].str.replace(".","")
df = df[~df['GPA_new'].str.isdigit() == False]
df['GPA'] = df['GPA'].astype(float)
df.drop('GPA_new',axis=1,inplace=True)

<ipython-input-12-dfa58d7c6482>:2: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  df['GPA_new'] = df['GPA'].str.replace(".","")


In [ ]:
df.head()

GPA  Gender  breakfast  calories_chicken  calories_day  calories_scone  \
0  2.400       2          1               430           1.0           315.0   
1  3.654       1          1               610           3.0           420.0   
2  3.300       1          1               720           4.0           420.0   
3  3.200       1          1               430           3.0           420.0   
4  3.500       1          1               720           2.0           420.0   

   coffee                      comfort_food        comfort_food_reasons  \
0       1                              none       we dont have comfort    
1       2       chocolate, chips, ice cream        Stress, bored, anger   
2       2   frozen yogurt, pizza, fast food             stress, sadness   
3       2  Pizza, Mac and cheese, ice cream                     Boredom   
4       2      Ice cream, chocolate, chips   Stress, boredom, cravings    

   comfort_food_reasons_coded  ...  soup  sports  thai_food tortilla_calories  \
0                         9.0  ...   1.0     1.0          1            1165.0   
1                         1.0  ...   1.0     1.0          2             725.0   
2                         1.0  ...   1.0     2.0          5            1165.0   
3                         2.0  ...   1.0     2.0          5             725.0   
4                         1.0  ...   1.0     1.0          4             940.0   

   turkey_calories  type_sports veggies_day  vitamins  waffle_calories  \
0              345   car racing           5         1             1315   
1              690  Basketball            4         2              900   
2              500         none           5         1              900   
3              690      Nothing           3         1             1315   
4              500     Softball           4         2              760   

                     weight  
0                       187  
1                       155  
2  I'm not answering this.   
3             Not sure, 240  
4                       190  

[5 rows x 61 columns]

In [ ]:
df.shape

(98, 61)

**Now we are left with 98 rows and 61 columns**

**Then, we use value_counts() function to check whether the column values are within the range of possible values.**

In [ ]:
df['Gender'].value_counts()

1    61
2    37
Name: Gender, dtype: int64

In [ ]:
df['breakfast'].value_counts()

1    88
2    10
Name: breakfast, dtype: int64

In [ ]:
df['calories_chicken'].value_counts()

610    49
720    25
430    19
265     5
Name: calories_chicken, dtype: int64

In [ ]:
df['calories_day'] = df['calories_day'].astype(int)
df['calories_day'].value_counts()

3    52
4    19
2    14
1    13
Name: calories_day, dtype: int64

In [ ]:
df['calories_scone'] = df['calories_scone'].astype(int)
df['calories_scone'].value_counts()

420    69
980    15
315    14
Name: calories_scone, dtype: int64

In [ ]:
df['coffee'].value_counts()

2    72
1    26
Name: coffee, dtype: int64

In [ ]:
df['comfort_food'].value_counts()

none                                                       1
Macaroni and cheese, chicken noodle soup, pizza            1
carrots, plantain chips, almonds, popcorn                  1
pizza, pasta, mac and cheese                               1
Chinese food, moes, sponge candy, homemade lasagne         1
                                                          ..
pizza, doughnuts, mcdonalds                                1
mac n cheese, peanut butter and banana sandwich, omelet    1
french fries, chips, ice cream                             1
chips and dip, pepsi,                                      1
Mac n Cheese, Lasagna, Pizza                               1
Name: comfort_food, Length: 98, dtype: int64

In [ ]:
df['comfort_food_reasons'].value_counts()

Boredom                                          3
Boredom                                          3
boredom                                          3
boredom, sadness                                 2
stress, boredom                                  2
                                                ..
Usually if I'm sad or depressed.                 1
No reasons                                       1
boredom, stress                                  1
Stress                                           1
happiness, they are some of my favorite foods    1
Name: comfort_food_reasons, Length: 87, dtype: int64

In [ ]:
df['comfort_food_reasons_coded'].value_counts()

2.0    36
1.0    21
9.0    19
3.0    12
7.0     3
5.0     3
4.0     2
6.0     1
8.0     1
Name: comfort_food_reasons_coded, dtype: int64

In [ ]:
df['cook'].value_counts()

3.0    39
2.0    28
1.0    12
4.0    12
5.0     7
Name: cook, dtype: int64

In [ ]:
df['cuisine'].value_counts()

1.0    71
6.0    14
2.0     9
4.0     2
3.0     1
5.0     1
Name: cuisine, dtype: int64

In [ ]:
df['diet_current'].value_counts()

eat good and exercise                                                                                                                                   1
I eat very basic foods like pizza and pasta. I don't try many new things.                                                                               1
i currently eat a lot of salad, but do not eat 3 times a day nor eat breakfast                                                                          1
I try to eat healthy but sometimes drink soda pop and I enjoy desserts. harder to eat healthy when I am at school.                                      1
Eat fruits and vegetables daily and with almost every meal. Diet mostly consists of meat as well.                                                       1
                                                                                                                                                       ..
I eat out more often then not. I try to make sure when I eat out it isn't fa

In [ ]:
df['diet_current_coded'].value_counts()

2    44
1    42
3    10
4     2
Name: diet_current_coded, dtype: int64

In [ ]:
df['drink'].value_counts()

2.0    54
1.0    44
Name: drink, dtype: int64

In [ ]:
df['eating_changes'].value_counts()

eat faster                                                                                                         1
Home cooked meals and a lot more difficult to have at college. It's typically quick, on the go foods.              1
I have been eating healthier especially vegetables and proteinous food                                             1
less healthy because of less options, money and time.                                                              1
Less snacking                                                                                                      1
                                                                                                                  ..
I do not snack as often at school. I find that I'm always busy so sometimes I don't have a lot of time to eat.     1
My diet is more limited just because of what I have time to make                                                   1
poor                                                            

In [ ]:
df['eating_changes_coded'].value_counts()

1    62
2    27
3     6
4     3
Name: eating_changes_coded, dtype: int64

In [ ]:
df['eating_changes_coded1'].value_counts()

3     37
5     24
2     13
4      7
8      4
10     3
11     3
7      2
1      1
6      1
9      1
13     1
12     1
Name: eating_changes_coded1, dtype: int64

In [ ]:
df['eating_out'].value_counts()

2    50
3    19
1    10
5    10
4     9
Name: eating_out, dtype: int64

In [ ]:
df['employment'].value_counts()

2.0    48
3.0    43
4.0     5
1.0     2
Name: employment, dtype: int64

In [ ]:
df['ethnic_food'].value_counts()

5    33
4    30
2    17
3    14
1     4
Name: ethnic_food, dtype: int64

In [ ]:
df['exercise'].value_counts()

1.0    49
2.0    32
3.0     9
5.0     8
Name: exercise, dtype: int64

In [ ]:
df['father_education'].value_counts()

4.0    37
2.0    25
5.0    21
3.0    12
1.0     3
Name: father_education, dtype: int64

In [ ]:
df['father_profession'].value_counts()

Dentist                              2
IT                                   2
Teacher                              2
Construction                         2
business owner                       2
                                    ..
Corporate Manager                    1
Biohemical Waste Elimination         1
Optometrist                          1
Owns his own promotional company     1
Store manager at Giant Eagle         1
Name: father_profession, Length: 92, dtype: int64

In [ ]:
df['fav_cuisine'].value_counts()

Italian                                     16
Italian                                     14
italian                                      6
Mexican                                      4
American                                     3
Chinese                                      3
American                                     3
Chinese                                      2
Asian                                        2
Mexican                                      2
italian                                      2
Thai                                         2
Korean                                       2
mac and cheese                               1
Vietnamese cuisine                           1
american                                     1
Japanese                                     1
I do not like cuisine                        1
Mexican cuisine                              1
Asian                                        1
Chinese food                                 1
Indian food  

In [ ]:
df['fav_cuisine_coded'].value_counts()

1    47
4    20
5    14
2     7
0     3
8     3
3     2
6     1
7     1
Name: fav_cuisine_coded, dtype: int64

In [ ]:
df['fav_food'].value_counts()

1.0    60
3.0    32
2.0     6
Name: fav_food, dtype: int64

In [ ]:
df['food_childhood'].value_counts()

Pizza                            3
Mac and cheese                   2
Steak                            2
Spaghetti                        2
Chicken                          2
                                ..
chicken parmesan                 1
Pasta                            1
pizza, pasta, grilled cheese     1
chicken, pasta, stir fry         1
pizza and spaghetti              1
Name: food_childhood, Length: 92, dtype: int64

In [ ]:
df['fries'].value_counts()

1    88
2    10
Name: fries, dtype: int64

In [ ]:
df['fruit_day'].value_counts()

5    56
4    25
3    13
2     3
1     1
Name: fruit_day, dtype: int64

In [ ]:
df['grade_level'].value_counts()

1    29
2    28
4    21
3    20
Name: grade_level, dtype: int64

In [ ]:
df['greek_food'].value_counts()

5    31
3    25
4    20
1    13
2     9
Name: greek_food, dtype: int64

In [ ]:
df['healthy_feeling'].value_counts()

5     12
7     12
8     12
4     11
3     11
2     10
6     10
9      9
1      6
10     5
Name: healthy_feeling, dtype: int64

In [ ]:
df['healthy_meal'].value_counts()

looks not oily                                                                                  1
Very colorful and smaller portions of the unhealthy food with larger portions of vegetables     1
Healthy meal for me is a food rich in protein, fiber, some sort of carbohydrates                1
lots of variety and veggies                                                                     1
Small portion of meat with majority fruits or vegetables                                        1
                                                                                               ..
a meal with a meat, vegetable, grain, and fruit                                                 1
Some kind of protein, a vegetable, and a grain                                                  1
a lot of greens                                                                                 1
4-6 ounces chicken or fish, side of potatoes, and green beans                                   1
A protein, a fruit, 

In [ ]:
df['ideal_diet'].value_counts()

being healthy                                                                                                        1
Ideally I would eat more fruits and vegetables in my diet along with some junk food.                                 1
it would be a vegetarian diet without any red meat                                                                   1
Eating healthy with lots of water and natural foods.                                                                 1
More organic food. Want to try and add more seafood into my diet.                                                    1
                                                                                                                    ..
I wish I had the time and energy to cook more at home on my own.                                                     1
Pretty much what I eat now                                                                                           1
healthy and protien                             

In [ ]:
df['ideal_diet_coded'].value_counts()

2    33
7    13
5    13
3    11
1    11
6    10
4     5
8     2
Name: ideal_diet_coded, dtype: int64

In [ ]:
df['income'].value_counts()

6.0    37
5.0    23
4.0    18
3.0    11
1.0     5
2.0     4
Name: income, dtype: int64

In [ ]:
df['indian_food'].value_counts()

5    28
3    26
1    20
4    12
2    12
Name: indian_food, dtype: int64

In [ ]:
df['italian_food'].value_counts()

5    79
4    13
3     6
Name: italian_food, dtype: int64

In [ ]:
df['life_rewarding'] = df['life_rewarding'].astype(int)
df['life_rewarding'].value_counts()

1     17
8     14
2     11
3     11
9     10
10     9
7      8
5      8
4      6
6      4
Name: life_rewarding, dtype: int64

In [ ]:
df['marital_status'] = df['marital_status'].astype(int)
df['marital_status'].value_counts()

1    51
2    47
Name: marital_status, dtype: int64

In [ ]:
df['meals_dinner_friend'].value_counts()

Chicken, Steak, Pasta                                     2
rice, chicken,  soup                                      1
Pasta, chicken, pizza                                     1
Rice with vegetables, chicken with pasta, salad           1
Pasta, pizza, and chicken                                 1
                                                         ..
spaghetti or pasta, shrimp fried rice, chicken            1
Some kind of pasta, a chicken dish, some kind of salad    1
pasta, lasagna, chicken                                   1
Tomato soup, Steak, crab                                  1
Pizza, Spaghetti, Baked Ziti                              1
Name: meals_dinner_friend, Length: 97, dtype: int64

In [ ]:
df['mother_education'] = df['mother_education'].astype(int)
df['mother_education'].value_counts()

4    38
2    23
5    19
3    15
1     3
Name: mother_education, dtype: int64

In [ ]:
df['mother_profession'].value_counts()

Teacher                                      3
Nurse                                        2
Banker                                       2
Secretary                                    2
unemployed                                   1
                                            ..
Works in retail                              1
stylist                                      1
management                                   1
Strategic Planning and Programs Manager      1
Receptionist for a medical supply company    1
Name: mother_profession, Length: 93, dtype: int64

In [ ]:
df['nutritional_check'].value_counts()

4    33
2    32
3    15
5    12
1     6
Name: nutritional_check, dtype: int64

In [ ]:
df['on_off_campus'] = df['on_off_campus'].astype(int)
df['on_off_campus'].value_counts()

1    78
2    15
3     4
4     1
Name: on_off_campus, dtype: int64

In [ ]:
df['parents_cook'].value_counts()

1    62
2    29
3     6
5     1
Name: parents_cook, dtype: int64

In [ ]:
df['pay_meal_out'].value_counts()

3    57
4    18
2    11
5     7
6     5
Name: pay_meal_out, dtype: int64

In [ ]:
df['persian_food'] = df['persian_food'].astype(int)
df['persian_food'].value_counts()

3    24
1    22
2    21
5    17
4    14
Name: persian_food, dtype: int64

In [ ]:
df['self_perception_weight'] = df['self_perception_weight'].astype(int)
df['self_perception_weight'].value_counts()

3    35
4    25
2    24
6     5
1     5
5     4
Name: self_perception_weight, dtype: int64

In [ ]:
df['soup'] = df['soup'].astype(int)
df['soup'].value_counts()

1    78
2    20
Name: soup, dtype: int64

In [ ]:
df['sports'] = df['sports'].astype(int)
df['sports'].value_counts()

1    64
2    34
Name: sports, dtype: int64

In [ ]:
df['thai_food'].value_counts()

5    29
4    21
3    19
1    16
2    13
Name: thai_food, dtype: int64

In [ ]:
df['tortilla_calories'] = df['tortilla_calories'].astype(int)
df['tortilla_calories'].value_counts()

1165    37
940     34
725     17
580     10
Name: tortilla_calories, dtype: int64

In [ ]:
df['turkey_calories'].value_counts()

500    39
690    34
345    17
850     8
Name: turkey_calories, dtype: int64

In [ ]:
df['type_sports'].value_counts()

Nothing                                                        11
none                                                            7
Hockey                                                          6
softball                                                        4
None                                                            4
Soccer                                                          3
Softball                                                        3
soccer                                                          3
Volleyball                                                      3
Wrestling                                                       2
Basketball                                                      2
Lacrosse                                                        2
Lacrosse                                                        2
Ice hockey                                                      2
field hockey                                                    2
hockey    

In [ ]:
df['veggies_day'].value_counts()

5    43
4    29
3    15
2     9
1     2
Name: veggies_day, dtype: int64

In [ ]:
df['vitamins'].value_counts()

1    49
2    49
Name: vitamins, dtype: int64

In [ ]:
df['waffle_calories'].value_counts()

1315    48
900     30
760     19
575      1
Name: waffle_calories, dtype: int64

In [ ]:
df['weight'].value_counts()

140                         7
150                         6
135                         6
190                         5
155                         5
175                         5
170                         5
185                         5
145                         4
130                         4
165                         4
125                         4
200                         3
180                         3
120                         3
129                         2
210                         2
160                         2
113                         2
167                         2
144 lbs                     1
260                         1
184                         1
192                         1
128                         1
138                         1
187                         1
169                         1
168                         1
205                         1
105                         1
123                         1
264                         1
110       

**Weight Column**

The weight column contains values in string format. To address this, let us use the isdigit() function to only keep the values that are digits.

In [ ]:
df = df[df['weight'].str.isdigit()]

In [ ]:
df['weight'].value_counts()

140    7
135    6
150    6
190    5
155    5
175    5
185    5
170    5
165    4
125    4
130    4
145    4
120    3
200    3
180    3
129    2
160    2
167    2
113    2
210    2
260    1
184    1
128    1
138    1
192    1
187    1
205    1
169    1
168    1
105    1
123    1
264    1
110    1
116    1
137    1
156    1
Name: weight, dtype: int64

In [ ]:
df.shape

(95, 61)

**Finally, we have 95 rows and 61 columns left**

In [ ]:
df.head()

GPA  Gender  breakfast  calories_chicken  calories_day  calories_scone  \
0  2.400       2          1               430             1             315   
1  3.654       1          1               610             3             420   
4  3.500       1          1               720             2             420   
5  2.250       1          1               610             3             980   
6  3.800       2          1               610             3             420   

   coffee                                  comfort_food  \
0       1                                          none   
1       2                   chocolate, chips, ice cream   
4       2                  Ice cream, chocolate, chips    
5       2                     Candy, brownies and soda.   
6       2  Chocolate, ice cream, french fries, pretzels   

                                comfort_food_reasons  \
0                              we dont have comfort    
1                               Stress, bored, anger   
4                         Stress, boredom, cravings    
5  None, i don't eat comfort food. I just eat whe...   
6                                    stress, boredom   

   comfort_food_reasons_coded  ...  soup  sports  thai_food tortilla_calories  \
0                         9.0  ...     1       1          1              1165   
1                         1.0  ...     1       1          2               725   
4                         1.0  ...     1       1          4               940   
5                         4.0  ...     1       2          4               940   
6                         1.0  ...     1       1          5               940   

   turkey_calories  type_sports veggies_day  vitamins  waffle_calories  weight  
0              345   car racing           5         1             1315     187  
1              690  Basketball            4         2              900     155  
4              500     Softball           4         2              760     190  
5              345        None.           1         2             1315     190  
6              690       soccer           4         1             1315     180  

[5 rows x 61 columns]

**Cleaned Data**

In [ ]:
pd.set_option('display.max_columns', None)
pd.set_option("display.max_rows", None)
df

GPA  Gender  breakfast  calories_chicken  calories_day  calories_scone  \
0    2.400       2          1               430             1             315   
1    3.654       1          1               610             3             420   
4    3.500       1          1               720             2             420   
5    2.250       1          1               610             3             980   
6    3.800       2          1               610             3             420   
7    3.300       1          1               720             3             420   
8    3.300       1          1               430             1             420   
9    3.300       1          1               430             3             315   
10   3.500       1          1               610             3             980   
11   3.904       1          1               720             4             420   
12   3.400       2          1               430             3             420   
13   3.600       1          1               610             3             420   
14   3.100       2          1               610             3             420   
16   4.000       1          1               265             3             420   
17   3.600       2          1               430             3             980   
18   3.400       1          1               720             3             980   
20   3.300       2          1               610             3             980   
22   3.700       2          1               610             3             420   
23   3.700       2          2               610             3             420   
24   3.900       1          1               720             2             420   
25   2.800       1          2               720             3             420   
26   3.700       2          1               610             2             420   
27   3.000       2          1               610             4             980   
28   3.200       2          1               610             2             420   
29   3.500       2          1               265             2             420   
31   4.000       2          1               610             3             420   
34   3.650       1          1               610             3             420   
35   3.000       1          1               610             2             420   
36   3.700       1          1               610             3             420   
37   3.400       1          1               720             4             420   
38   3.890       1          1               610             3             980   
39   3.000       2          1               720             1             980   
40   3.400       2          1               430             3             315   
41   2.900       1          1               720             4             980   
42   3.600       1          1               610             3             420   
44   3.200       1          1               610             4             420   
45   3.605       1          1               610             3             315   
46   3.800       2          1               430             2             420   
47   2.800       2          1               430             1             980   
48   3.500       2          2               430             1             315   
50   3.600       2          1               720             3             420   
51   3.300       2          1               610             4             980   
52   3.300       2          1               610             4             420   
54   3.500       2          1               610             3             420   
55   3.350       1          2               610             2             315   
56   3.800       2          1               720             4             315   
57   2.800       1          1               610             4             980   
58   3.500       1          1               610             3             420   
59   3.700       1          1         

In [ ]:
df.to_excel("cleaned_lab1.xlsx")